tests/unit/type/zset.tcl

```diff
@@ -1516,11 +1517,12 @@ start_server {tags {"zset"}} {
 
             assert_encoding $encoding zscoretest
             for {set i 0} {$i < $elements} {incr i} {
-                assert_equal [lindex $aux $i] [r zscore zscoretest $i]
+                assert_equal [format %.17g [lindex $aux $i]] [format %.17g [r zscore zscoretest $i]]
```

Comment:
please explain why this formatting is needed.
maybe in a code comment as well.

Reply:
@oranagra fixed it in the latest commit. Please review

Reply:
that format is used in 3 places, so the comment may be needed in all, or at least some cross reference.
but i'm still not sure i understand what changed here in that respect and why we needed that change, can you please explain

Reply:
@oranagra WRT to the 3 places I've added a note on the FP representation and made reference on all 3.
WRT to why we needed that change it was because snprintf could output more digits than precision digits required to represent the number ( see https://www.ncl.ucar.edu/Support/talk_archives/2013/2176.html ). To ensure that there is no difference I'm making sure that we use the same format to compare the string representation instead of the doubles. Notice both snprintf and the grisu2 produce IEEE compliant representations.

Reply:
Does `assert_equal` compare the values as strings or as floats?

If we compare the string representation, I agree they can differ for the same float value, but the float values should be equal since there are no float calculations, thus no rounding errors involved.

    % expr 0.8 == 0.79999999999999999
    1
    % expr 0.8 == 0.8e0
    1
    % expr 0.79999999999999999
    0.8

Therefore, I think we can compare them as floats and expect them to be equal. We can use `expr` to convert a string to float.

Another way is of course to convert the string to float and then back to string using `format %.17g` but it seems slightly more complex to me.

Btw, an implicit conversion from float to string seems to give 17 digits precision in TCL 8.6 anyway:

```
$ tclsh
% expr 0.12345678901234567890
0.12345678901234568
% puts [expr 0.12345678901234567890]
0.12345678901234568
```

Reply:
@oranagra WRT 

> The code comments may be OK, my recent questions about the test suite changes are if you can explain here why these changes are needed.
> I.e. If the new algorithm was completely compatible, no changes would be required. So I want to understand the differences.

Here's a sample of the failed test:  
```
Expected '0.62137966259446908' to be equal to '0.6213796625944691'
```

Notice that the comparisons are string. If we convert them to double they represent the same number as showcased bellow:
```
:~# tclsh
% expr 0.62137966259446908
0.6213796625944691
% expr 0.6213796625944691
0.6213796625944691
%
```
Notice that the last string is the output of the grisu2, meaning we're replying in the exact IEEE required format representation. 

@zuiderkwast you're indeed right. I was converting reply to double and then back to string with the format specified. I've changed the comparison towards your recommendation and there is no need to do the conversion.

Reply:
Nice that you agree. :-)